pandas/core/indexes/interval.py

```diff
@@ -1092,7 +1092,15 @@ def overlaps(self, other):
 
     def _setop(op_name):
         def func(self, other, sort=True):
-            other = self._as_like_interval_index(other)
+            try:
+                other = self._as_like_interval_index(other)
```

Comment:
It looks like `_as_like_interval_index` is only used here.  Perhaps just including that logic here could help clean things up?  Not entirely sure though.

Reply:
agreed and done, also allows me to get rid of the try except that really should have just been an if condition.